devops/scripts/benchmarks/benches/gromacs.py

```diff
@@ -0,0 +1,248 @@
+            # "0384",
+        ]
+        benches_pme = [GromacsBenchmark(self, model, "pme") for model in models_pme]
+        # Add more models as needed
+
+        return benches_rf + benches_pme
+
+    def setup(self):
+        self.gromacs_src = git_clone(
+            self.directory,
+            "gromacs-repo",
+            self.git_url(),
+            self.git_tag(),
+        )
+
+        # Build GROMACS
+        run(
+            [
+                "cmake",
+                f"-S {str(self.directory)}/gromacs-repo",
+                f"-B {self.gromacs_build_path}",
+                f"-DCMAKE_BUILD_TYPE=Release",
+                f"-DCMAKE_CXX_COMPILER=clang++",
+                f"-DCMAKE_C_COMPILER=clang",
+                f"-DGMX_GPU=SYCL",
+                f"-DGMX_SYCL_ENABLE_GRAPHS=ON",
+                f"-DGMX_FFT_LIBRARY=MKL",
+                f"-DGMX_BUILD_OWN_FFTW=ON",
+                f"-DGMX_GPU_FFT_LIBRARY=MKL",
+                f"-DGMX_GPU_NB_CLUSTER_SIZE=8",
+                f"-DGMX_OPENMP=OFF",
+            ],
+            add_sycl=True,
+        )
+        run(
+            f"cmake --build {self.gromacs_build_path} -j {options.build_jobs}",
+            add_sycl=True,
+        )
+        self.download_and_extract_grappa()
+
+    def download_and_extract_grappa(self):
+        grappa_tar_file = self.directory / self.grappa_file()
+
+        model = download(
+            self.directory,
+            self.grappa_url(),
+            grappa_tar_file,
+            checksum="cc02be35ba85c8b044e47d097661dffa8bea57cdb3db8b5da5d01cdbc94fe6c8902652cfe05fb9da7f2af0698be283a2",
+            untar=True,
+        )
+        if options.verbose:
+            print(f"Grappa tar file downloaded and extracted to {model}")
+
+    def teardown(self):
+        pass
+
+
+class GromacsBenchmark(Benchmark):
+    def __init__(self, suite, model, type):
+        self.suite = suite
+        self.model = model  # The model name (e.g., "0001.5")
+        self.type = type
+        self.gromacs_src = suite.gromacs_src
+        self.grappa_dir = suite.grappa_dir
+        self.gmx_path = suite.gromacs_build_path / "bin" / "gmx"
+
+    def name(self):
+        return f"gromacs-{self.model}-{self.type}"
+
+    def setup(self):
+        model_dir = self.grappa_dir / self.model
+        if self.type != "rf" and self.type != "pme":
+            raise ValueError(f"Unknown benchmark type: {self.type}")
+
+        cmd_list = [
+            str(self.gmx_path),
+            "grompp",
+            "-f",
+            f"{str(self.grappa_dir)}/{self.type}.mdp",
+            "-c",
+            str(model_dir / "conf.gro"),
+            "-p",
+            str(model_dir / "topol.top"),
+            "-o",
+            f"{str(model_dir)}/{self.type}.tpr",
+        ]
+
+        # Generate configuration files
+        self.conf_result = run(
+            cmd_list,
+            add_sycl=True,
+        )
+
+    def run(self, env_vars):
+        if not self.gmx_path.exists():
```

Comment:
Move this check to setup() as you try running this binary there first without a check.